PyStemmusScope/docker_process.py

```diff
@@ -0,0 +1,112 @@
+"""The Docker STEMMUS_SCOPE model process wrapper."""
+from PyStemmusScope.config_io import read_config
+from pathlib import Path
+import os
+import docker
+
+
+def make_docker_vols_binds(cfg_file: str) -> tuple[list[str], list[str]]:
+    """Make docker volume mounting configs.
+
+    Args:
+        cfg_file: Location of the config file
+
+    Returns:
+        volumes, binds
+    """
+    cfg = read_config(cfg_file)
+    
+    volumes = [cfg["OutputPath"], cfg["InputPath"]]
+    binds = [
+        f"{cfg['OutputPath']}:{cfg['OutputPath']}:rw",
+        f"{cfg['InputPath']}:{cfg['InputPath']}:ro",
+    ]
+
+    if (
+        not Path(cfg_file).parent.is_relative_to(cfg["InputPath"]) or
```

Comment:
```suggestion
        not Path(cfg_file).parent.is_relative_to(cfg["InputPath"]) and
```

Reply:
Assume that config file is located in the `InputPath`, if `or` is used in the statement, the `InputPath` is mounted two times and `docker.client` returns error `APIError: 400 Client Error for : Bad Request ("Duplicate mount point: ...)`

Reply:
Well spotted, I wrote this if statement  to prevent that error from occurring, but messed it up :man_facepalming:

Reply:
I refactored this to be more accurate and robust.